### PR TITLE
[opencl] Add convolutiongrad support

### PR DESCRIPTION
**Pull request by opti-mix**
This operation is necessary for the training mode. It is not optimized for speed yet.

The implementation is making use of atomics to ensure the correctness of the results when this kernel is executed by multiple threads on GPUs.

### CODE REVIEW

File: lib/Backends/OpenCL/kernels.cl [hunk 1]
```diff
@@ -15,6 +15,30 @@ typedef struct {
   cl_uint64_t c; // Number of channels
 } ShapeNHWC;
 
+#if defined(cl_khr_int32_base_atomics)
```
Comment (rdzhabarov) [inline lib/Backends/OpenCL/kernels.cl]: can you do some indentation for readability?

File: tests/unittests/OCLTest.cpp [hunk 2]
```diff
@@ -103,6 +103,37 @@ TEST(OpenCLCorrectnessTest, softmaxGradTest) {
...
 }
 
+TEST(OpenCLCorrectnessTest, convGradTest) {
+  Tensor inputs(ElemKind::FloatTy, {9, 8, 9, 4});
+  Tensor kernel1(ElemKind::FloatTy, {3, 3, 3, 4});
+  Tensor bias1(ElemKind::FloatTy, {3});
+  Tensor kernel2(ElemKind::FloatTy, {2, 2, 2, 1});
+  Tensor bias2(ElemKind::FloatTy, {2});
+  Tensor selected(ElemKind::IndexTy, {9, 1});
+  inputs.getHandle().initXavier(1);
```
Comment (rdzhabarov) [inline tests/unittests/OCLTest.cpp]: nit: I think typically xavier initialization is applied to the weights/bias or other learning params.

Comment (opti-mix) [reply tests/unittests/OCLTest.cpp]: The test is copied 1:1 from JITTest::convGradTest written by @hegemanjwh2.

@hegemanjwh2 Do you remember what was the intent of this Xavier initialization?

File: lib/Backends/OpenCL/kernels.cl (hunk 1 above)
Comment (opti-mix) [reply lib/Backends/OpenCL/kernels.cl]: Indented the atomicAdd function. 
And preprocessor directives are usually not indented, or?

**Pull request merged**